stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/StringExtensions.kt

```diff
@@ -57,3 +65,150 @@ public fun String.cidToTypeAndId(): Pair<String, String> {
     check(':' in this) { "cid needs to be in the format channelType:channelId. For example, messaging:123" }
     return checkNotNull(split(":").takeIf { it.size >= 2 }?.let { it.first() to it.last() })
 }
+
+/**
+ * Returns [StreamCdnOriginalImageDimensions] if the image is hosted by Stream's CDN and is resizable,
+ * otherwise returns null.
+ *
+ * @return Class containing the original width and height dimensions of the image or null.
+ */
+public fun String.getStreamCdnHostedImageDimensions(): StreamCdnOriginalImageDimensions? {
+    return try {
+        val width = this.substringAfter(delimiter = "ow=", missingDelimiterValue = "")
+            .takeIf { it.isNotBlank() }
+            ?.substringBefore("&")
+            ?.toInt()
+
+        val height = this.substringAfter(delimiter = "oh=", missingDelimiterValue = "")
+            .takeIf { it.isNotBlank() }
+            ?.substringBefore("&")
+            ?.toInt()
+
+        if (height != null && width != null) {
+            StreamCdnOriginalImageDimensions(
+                originalWidth = width,
+                originalHeight = height
+            )
+        } else {
+            null
+        }
+    } catch (e: java.lang.Exception) {
+        val logger = StreamLog.getLogger("Chat: getStreamCDNHostedImageDimensions")
+        logger.e { "Failed to parse Stream CDN image dimensions from the URL:\n ${e.stackTraceToString()}" }
+
+        null
+    }
+}
+
+/**
+ * Generates a string URL with Stream CDN image resizing query parameters added to it. Once this URL is called, Stream's
+ * CDN will generate a resized image which is accessible using the link returned by this function.
+ *
+ * @param resizedWidthPercentage The percentage of the original image width the resized image width will be.
+ * @param resizedHeightPercentage The percentage of the original image height the resized image height will be.
+ * @param resizeMode Sets the image resizing mode. The default mode is [StreamCdnResizeImageMode.CLIP].
+ * @param cropMode Sets the image crop mode. The default mode is [StreamCdnCropImageMode.CENTER].
+ */
+public fun String.createResizedStreamCdnImageUrl(
```

Comment:
Would be great to be able to test this properly, but unforunatelly `Uri` is an Android class and can't easily be covered by unit tests.

I don't think mocking it is the answer since it's responsible for core behavior which we wouldn't want to get wrong in a mock.

Reply:
It can be unit-tested with robolectric: https://stackoverflow.com/a/46317695

Reply:
Suggestion was implemented in the following commit: https://github.com/GetStream/stream-chat-android/commit/873f11fe673e0694b7551cfc59c23fffbcb13432